packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -470,7 +470,15 @@ export class DataStores implements IDisposable {
 		requestHeaderData: RuntimeHeaderData,
 	): Promise<FluidDataStoreContext> {
 		const headerData = { ...defaultRuntimeHeaderData, ...requestHeaderData };
-		if (this.checkAndLogIfDeleted(id, "Requested", "getDataStore", requestHeaderData)) {
+		if (
+			this.checkAndLogIfDeleted(
+				id,
+				this.contexts.get(id),
```

Comment:
So below it uses `getBoundOrRemoted`, will that wait or something, whereas this just takes whatever is here now?  Does this difference matter?

Reply:
`getBoundOrRemoted` will wait for a data store is it doesn't exist if `wait` is set. We cannot use that when checking for deleted because it will wait forever for a deleted data store.
So, using this.contexts.get(id) is good enough to be used here since we are only using it in scenarios where a data store has been deleted.